Set dtype when copying empty tensor

huydhn opened the pull request:
Summary: Forward fix D57251348

Test Plan: `buck2 test 'fbcode//mode/dev' fbcode//executorch/kernels/test:aten_op_copy_test`

Differential Revision: D57304360

huydhn commented:
Plz let me know if there are more tests that can be added here instead of relying on the one from ET

malfet commented on aten/src/ATen/native/Copy.cpp:
```diff
@@ -325,7 +325,7 @@ Tensor copy(const Tensor& self, const Tensor& src, bool non_blocking) {
   // that copy_() will fully overwrite all data with that of src
   if (self_storage->nbytes() == 0) {
-    r = at::empty_strided(self.sizes(), self.strides());
+    r = at::empty_strided(self.sizes(), self.strides(), self.dtype());
```
Do you need to preserve device here as well? and memory layout?

huydhn replied:
I think the correct behavior comes from https://github.com/pytorch/pytorch/blob/main/aten/src/ATen/native/TensorShape.cpp#L3978, so it would be 
 `torch.preserve_format` https://pytorch.org/docs/stable/tensor_attributes.html#torch.memory_format, and yes to both these questions.

Waiting for @bdhirsh to confirm.

bdhirsh replied:
@huydhn ah yep, we should preserve both of them (you can do that by passing in the `TensorOptions` of self):
```
at::empty_strided(self.sizes(), self.strides(), self.options());
```

bdhirsh reviewed (approved):
Thank you for the forward fix!

huydhn committed:
Set dtype when copying empty tensor (#126124)

Summary:
Pull Request resolved: https://github.com/pytorch/pytorch/pull/126124

Forward fix D57251348

Test Plan: `buck2 test 'fbcode//mode/dev' fbcode//executorch/kernels/test:aten_op_copy_test`

Reviewed By: bdhirsh

Differential Revision: D57304360
